### PR TITLE
AutodiffCostFunction now expands tensors with batch size 1 before running vmap

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

File: theseus/core/cost_function.py [hunk 1]
```diff
@@ -240,6 +241,17 @@ def _compute_autograd_jacobian_vmap(
...
+            tensors: Tuple[torch.Tensor, ...], batch_size: int
+        ) -> Tuple[torch.Tensor, ...]:
+            return tuple(
+                t if t.shape[0] == batch_size else t.expand((batch_size,) + t.shape[1:])
+                for t in tensors
+            )
+
+        batch_size = max(t.shape[0] for t in chain(optim_tensors, aux_tensors))
+        optim_tensors = _expand_all(optim_tensors, batch_size)
+        aux_tensors = _expand_all(aux_tensors, batch_size)
```
Comment (fantaosha) [inline theseus/core/cost_function.py:254]: I assume this is more related with vectorization. Do we need to guarantee there are no multiple `batch_size!=1`?

**Review by fantaosha — COMMENTED** [review]
LGTM overall.

**Review by fantaosha — APPROVED** [review]
LGTM.

**Review by mhmukadam — APPROVED** [review]
LGTM.

**Pull request merged**